### PR TITLE
rename SealCredPaymaster to BWLPaymaster

### PR DESCRIPTION
**Pull request by MixailE**

### CODE REVIEW

**Comment by MixailE** [comment]
[Rename `seal-cred-paymaster-contract` to `bwl-paymaster-contract`](https://trello.com/c/ddCIHrgA/1669-rename-seal-cred-paymaster-contract-to-bwl-paymaster-contract)

File: hardhat.config.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ const config: HardhatUserConfig = {
   networks: {
     deploy: {
       url: ETH_RPC,
+      //       url: 'https://polygon-mumbai.infura.io/v3/f605b671970744d3a688de67dc8c2323',
```
Comment (smyzin) [inline hardhat.config.ts]: Let's remove the comment

File: package.json [hunk 2]
```diff
@@ -68,5 +67,8 @@
     "typechain": "^8.1.1",
     "typescript": "^4.9.3"
   },
-  "packageManager": "yarn@3.3.0"
+  "packageManager": "yarn@3.3.0",
+  "dependencies": {
+    "@big-whale-labs/constants": "^0.1.73"
+  }
```
Comment (smyzin) [inline package.json]: It should be above `devDependencies`

Comment (smyzin) [reply package.json]: Also, do we need to move in from `devDependencies` to `dependencies`?

File: scripts/deploy.ts [hunk 3]
```diff
@@ -92,9 +92,9 @@ async function main() {
   console.log('Contract address:', address)
   console.log(
     'Etherscan URL:',
-    `https://${
-      chainName !== 'mainnet' ? `${chainName}.` : ''
-    }etherscan.io/address/${address}`
+    `https://${chainName !== 'mainnet' ? `${chainName}.` : ''}${
+      chainId === 80001 ? 'polygonscan.com' : 'etherscan.io'
```
Comment (smyzin) [inline scripts/deploy.ts]: I think it won't work if we use `polygon mainnet`, will it?

**Review by smyzin — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (T-Damer) [reply package.json]: We probably can have it anywhere we want

File: scripts/deploy.ts [hunk 4]
```diff
@@ -92,9 +92,9 @@ async function main() {
   console.log('Contract address:', address)
   console.log(
     'Etherscan URL:',
-    `https://${
-      chainName !== 'mainnet' ? `${chainName}.` : ''
-    }etherscan.io/address/${address}`
+    `https://${chainName !== 'mainnet' ? `${chainName}.` : ''}${
```
Comment (T-Damer) [inline scripts/deploy.ts]: Let's extract this into separate function, also, change the check to be:
```TS
${chainName === 'mainnet' ? '' : chainName}
```
We don't need double nested `${${}}` here

File: scripts/deploy.ts (hunk 3 above)
Comment (T-Damer) [reply scripts/deploy.ts]: Probably, but we may need paymaster on polygon as well, so don't change this

File: scripts/deploy.ts [hunk 5]
```diff
@@ -29,28 +28,30 @@ async function main() {
     3: 'ropsten',
     4: 'rinkeby',
     5: 'goerli',
+    80001: 'mumbai',
   } as { [chainId: number]: string }
   const chainName = chains[chainId]
 
-  const contractName = 'SealCredPaymaster'
+  const contractName = 'BWLPaymaster'
```
Comment (T-Damer) [inline scripts/deploy.ts:51]: Let's extract this, so you don't need to change it here and in `test/BWLPaymaster.ts` each time

Comment (T-Damer) [reply scripts/deploy.ts:51]: Something like `data/contractName.ts`

**Review by T-Damer — APPROVED** [review]

File: scripts/deploy.ts (hunk 5 above)
Comment (MixailE) [reply scripts/deploy.ts:51]: This is not necessary, we have a name in the contract and we still have to change it

**Pull request merged**